earlydecoder/stacks/decoder.go

```diff
@@ -52,21 +50,13 @@ func LoadStack(path string, files map[string]*hcl.File) (*stack.Meta, hcl.Diagno
 		var err error
 		src, err = tfaddr.ParseProviderSource(req.Source)
 		if err != nil {
-			diags = append(diags, &hcl.Diagnostic{
-				Severity: hcl.DiagError,
-				Summary:  fmt.Sprintf("Unable to parse provider source for %q", name),
-				Detail:   fmt.Sprintf("%q provider source (%q) is not a valid source string", name, req.Source),
-			})
+			// This is handled in decodeRequiredProvidersBlock now
```

Comment:
Since we're already doing the conversion in `decodeRequiredProvidersBlock`, can we store the parsed value instead of the string? This would allow us to get rid of this loop entirely